src/module-elasticsuite-indices/Helper/Index.php

```diff
@@ -0,0 +1,306 @@
+     */
+    private const SECONDS_IN_DAY = 86400;
+
+    /**
+     * Store number of seconds in a year.
+     */
+    private const SECONDS_IN_YEAR = 31536000;
+
+    /**
+     * @var array
+     */
+    private $elasticSuiteIndices = [
+        'catalog_category',
+        'catalog_product',
+        'thesaurus',
+    ];
+
+    /**
+     * @var Client
+     */
+    private $esClient;
+
+    /**
+     * @var DataObject
+     */
+    protected $workingIndexers;
+
+    /**
+     * @var DataObject
+     */
+    protected $storeIndices;
+
+    /**
+     * @var IndexSettings
+     */
+    private $indexSettings;
+
+    /**
+     * PHP Constructor
+     *
+     * @param Context                         $context                  The current context.
+     * @param Client                          $esClient                 ElasticSearch client.
+     * @param WorkingIndexerCollectionFactory $indexerCollectionFactory Working indexers collection.
+     * @param StoreIndicesCollectionFactory   $storeIndicesFactory      Store indices collection.
+     * @param IndexSettings                   $indexSettings            ElasticSuite index settings.
+     */
+    public function __construct(
+        Context $context,
+        Client $esClient,
+        WorkingIndexerCollectionFactory $indexerCollectionFactory,
+        StoreIndicesCollectionFactory $storeIndicesFactory,
+        IndexSettings $indexSettings
+    ) {
+        parent::__construct($context);
+        $this->esClient = $esClient;
+        $this->workingIndexers = $indexerCollectionFactory->create()->getItems();
+        $this->storeIndices = $storeIndicesFactory->create()->getItems();
+        $this->indexSettings = $indexSettings;
+    }
+
+    /**
+     * Get ElasticSuite indices.
+     *
+     * @param array $params Parameters array.
+     * @return array
+     * @throws \Exception
+     */
+    public function getElasticSuiteIndices($params = []): array
+    {
+        $elasticSuiteIndices = [];
+
+        foreach ($this->esClient->getIndexAliases($params) as $name => $aliases) {
+            if ($this->isElasticSuiteIndex($name)) {
+                $elasticSuiteIndices[$name] = $aliases ? key($aliases['aliases']) : null;
+            }
+        }
+
+        return $elasticSuiteIndices;
+    }
+
+    /**
+     * Returns if index is elastic suite index.
+     *
+     * @param string $indexName Index name.
+     * @return bool
+     */
+    public function isElasticSuiteIndex($indexName): bool
+    {
+        foreach ($this->elasticSuiteIndices as $elasticSuiteIndex) {
+            if (strpos($indexName, $elasticSuiteIndex) !== false) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Get a index status.
+     *
+     * @param string $indexName Index name.
+     * @param string $alias     Index alias.
+     *
+     * @return string
+     */
+    public function getIndexStatus($indexName, $alias): string
```

Comment:
`Helper` containing many method is a bit old-school and should be refactored.

Most probably this method could be into a single-purpose object called something like "IndexStatusProvider".